### DISCUSSION TITLE
NSPanel Pro Version Update 3.0.0

### DISCUSSION THREAD

**Discussion by dpozimski** [seaky/nspanel_pro_tools_apk#83] [Ideas]
According to latest release of Sonoff NSPanel Pro I wonder to ask if someone are able to grab the OTA update and upload it to the
https://drive.google.com/drive/folders/1bHLLJy8vYUjkCCnyUmrHgEFWi_2cxywv ?

Is there another way to upgrade the rooted NSPanel without waiting for a file? Is it possible to restore offical updates from sonoff after activation of developer mode (adb) ?

**Comment by bobzer** [comment]
would be interest in the cast app for the 2 way call, but the rest ? since we don't use their dashboard but ours, I don't see the value ?

**Comment by seaky** [comment]
You will find the v3.0.0 on the reffered link. Somebody uploaded :)

btw I compared the 3.0.0 to the 1.5.6 and 
- the OS build number and patch level is the same: OPM8.190605.003, 2019-07-05
- majority of files are binary identical to 1.5.6 and 2.3.0

changes:
- new bcm4345c0 bluetooth firmware
- new zgateway client which includes new matter integrations
- new eWeLink application v3.0.0

I'll make firmware independent zgatway package which will be included in the new version of my app.

I haven't yet checked the root capability...

**Reply by dpozimski** [reply]
Thank you very much! I am using the your tool to use panel with Xiaomi camera app and EWeLink control app (it has better look than my HA dashboard)

**Comment by StefanHesse** [comment]
I got my panel 2 days ago and was able to install all necessary stuff, but neither with this tool nor with Automagic 1.38 i get the proximity sensor doing what i want.
I can configure it (the difference is i have to define a value of 2000 and not 300 like it is shown in many videos to have a proximity of ra. 20cm), but it is never waking up my device.
Is it me or is there a bug in the NSPanel 3.0.0 which my device updated to before i hacked it?

**Reply by seaky** [reply]
What is the idle value? (without anything in front if it)

**Reply by StefanHesse** [reply]
The idle value is 1950 and it's going up until 60000 if i put my hand direct in front of the glass nearly touching it

**Comment by jengel84** [comment]
@StefanHesse 
I just installed two panels this week with 3.0. it works.

My findings: I had to set totally different values for the proximity sensor on each (400 vs 4700). I checked the sensor output to find the values. Now it works as expected.

Any experiences with only dimming the screen instead of sleep? Will there be burn-in or similar effects?

**Reply by StefanHesse** [reply]
I have no idea why 2000 did not work, but I now configured the value to 3000 and it seems to be working this way.
I have to fine tune a little bit more to get this more to the proximity i want, but different devices seem to have totally different values.

**Reply by jengel84** [reply]
Thanks for sharing that knowledge, @seaky .

Excellent work btw. This really improves the WAF of my integration tremendously.

**Comment by seaky** [comment]
idle 1950 is fully normal. It depends on the calibrated value. 
what are very important: 
- always set a clearly distinguishable value. For example if the idle is 2000, 3000 is the minimum. 
- this kind of sensors are designed to sense proximity not more than 5-8cm so never try to set it up to 15-20cm it wont work reliabily
